engine/access/rpc/backend/backend_scripts.go

```diff
@@ -14,12 +16,16 @@ import (
 	"github.com/onflow/flow-go/storage"
 )
 
+// uniqueScriptLoggingTimeWindow is the duration for checking the uniqueness of scripts sent for execution
+const uniqueScriptLoggingTimeWindow = 10 * time.Minute
+
 type backendScripts struct {
 	headers           storage.Headers
 	executionReceipts storage.ExecutionReceipts
 	state             protocol.State
 	connFactory       ConnectionFactory
 	log               zerolog.Logger
+	seenScripts       map[[md5.Size]byte]time.Time // to keep track of unique scripts sent by clients
```

Comment:
maybe add a comment here that the max memory size of this slice is bounded to `1MB (2^16*2*8)`